### DISCUSSION TITLE
v.7 example?

### DISCUSSION THREAD

**Discussion by proko** [davidjerleke/embla-carousel#638] [Help] [answered]
Hi there! As far as I understand all docs and examples use v8rc is that right? Is there any v7 example to have a look at? 
In my daywork I can't use v8, however v7 doesn't seem to work as expeted with the apis in the docs.

**Comment by davidjerleke** [comment, marked as answer]
Hi @proko,

> As far as I understand all docs and examples use v8rc is that right? 

That's correct. The docs is for the latest release which is `v8.0.0-rc14` at the time of writing.

> Is there any v7 example to have a look at?

No unfortunately not. I'm the sole dev in this project so I don't have the time to maintain multiple documentation versions. But what you could do is that you can clone the repo, checkout to an ancient commit where v7 is the latest and run the `embla-carousel-docs` dev command locally. 

Best,
David

**Reply by proko** [reply]
Thanks @davidjerleke, I'll try the cloning thing!!

**Reply by proko** [reply]
So unfortunately this didn't work. Getting an error about dependencies 
```
ModuleNotFoundError: Module not found: Error: Can't resolve 'embla-carousel-react' in '.../packages/embla-car
  ousel-docs/src/components/CodeSandbox/React/SandboxFilesSrc/Thumbs'
```

Tried to figure out how a React initialisation is supposed to work by looking directly into the source code but couldn't read it there.
I'm doing the init the way it is described in v8 docs. My problem is that in the first render of a looped carousel the last slides overlap the first one. So it looks like the transform is not calculated accurately. If I force a rerender however (by resizing the window), on the next cycle it does place the slides correctly.
Does it ring any bells? I'd really appreciate it if you could give an example of how initialisation should look like in v7.

**Comment by proko** [comment]
Regardless of my problem. Wouldn't actually make more sense to keep v7 docs live instead of pretty much temporary docs of an RC version? This makes adoption of the library extremely difficult by any organisation.

**Reply by davidjerleke** [reply]
Hi again @proko,

> Wouldn't actually make more sense to keep v7 docs live instead of pretty much temporary docs of an RC version? This makes adoption of the library extremely difficult by any organisation.

**No.** An RC doesn't always mean that a product is unstable. At this point, the **RC** is `v15` and it's very stable. Actually, I would argue that it's more stable than `v7`. One or two serious bugs that was in `v7` have been solved and there's also one major change that makes the RC more stable than `v7`:

- #589

The test have been refactored from scratch and are much more reliable now. While I wrote the new tests, bugs from `v7` surfaced which I was able to solve once discovered. 

However, I think your **problem** is [related to this](https://github.com/davidjerleke/embla-carousel/issues/159#issuecomment-805024840). It's worth noting that `v8` solves this automatically.
 
Best,
David

**Reply by proko** [reply]
Thanks again @davidjerleke for the response and everything you do for the community! Much appreciated. We did go with rc15 eventually. Looking forward to knowing what's next for embla.
